tests/subsurfaces.cpp

```diff
@@ -26,16 +26,103 @@
 
 using namespace testing;
 
-class SubsurfaceTest : public wlcs::StartedInProcessServer
+struct AbstractInputDevice
+{
+    virtual void down_at(int x, int y) = 0;
+    virtual wl_surface* focused_window() = 0;
+    virtual std::pair<int, int> position() = 0;
+    virtual ~AbstractInputDevice() = default;
+};
+
+struct PointerInputDevice : AbstractInputDevice
+{
+    PointerInputDevice(wlcs::Server& server, wlcs::Client& client):
+        client{client},
+        pointer{server.create_pointer()}
+    {
+    }
+
+    void down_at(int x, int y) override
```

Comment:
The name doesn't seem to match the implementation. Which is right?

Reply:
It's an override of an abstract method. 'down_at' doesn't make sense for pointers and 'move_to' doesn't make sense for touch. Maybe there's a name out there that's applicable to both, but I just choose one and went with it.

Reply:
place_at()?

Reply:
Maybe `to_screen_position` is a bit better